solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3577 @@
+            })
+
+            it("should emit a MaintainerUnauthorized event", async () => {
+              await expect(tx8)
+                .to.emit(maintainerProxy, "MaintainerUnauthorized")
+                .withArgs(maintainer8.address)
+            })
+          }
+        )
+      })
+    })
+  })
+
+  describe("updateBridge", () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    context("when called by a third party", () => {
+      it("should revert", async () => {
+        await expect(
+          maintainerProxy.connect(thirdParty).updateBridge(thirdParty.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+        tx = await maintainerProxy
+          .connect(governance)
+          .updateBridge(thirdParty.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should update the bridge", async () => {
+        expect(await maintainerProxy.bridge()).to.equal(thirdParty.address)
+      })
+
+      it("should emit the BridgeUpdated event", async () => {
+        await expect(tx)
+          .to.emit(maintainerProxy, "BridgeUpdated")
+          .withArgs(thirdParty.address)
+      })
+    })
+  })
+
+  describe("updateGasOffsetParameters", () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    context("when called by a third party", () => {
+      it("should revert", async () => {
+        await expect(
+          maintainerProxy
+            .connect(thirdParty)
+            .updateGasOffsetParameters(
+              40,
+              41,
+              42,
+              43,
+              44,
+              45,
+              46,
+              47,
+              48,
+              49,
+              50,
+              51
+            )
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+        tx = await maintainerProxy
+          .connect(governance)
+          .updateGasOffsetParameters(
+            40,
+            41,
+            42,
+            43,
+            44,
+            45,
+            46,
+            47,
+            48,
+            49,
+            50,
+            51
+          )
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should emit the GasOffsetParametersUpdated event", async () => {
+        await expect(tx)
+          .to.emit(maintainerProxy, "GasOffsetParametersUpdated")
+          .withArgs(40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51)
+      })
+    })
```

Comment:
We need to test if they were actually updated. That's why we need to make them public or have a function allowing to fetch their current values.

Reply:
done